Library/Services/INfieldSurveyManualTestsService.cs

```diff
@@ -0,0 +1,49 @@
+﻿//    This file is part of Nfield.SDK.
+//
+//    Nfield.SDK is free software: you can redistribute it and/or modify
+//    it under the terms of the GNU Lesser General Public License as published by
+//    the Free Software Foundation, either version 3 of the License, or
+//    (at your option) any later version.
+//
+//    Nfield.SDK is distributed in the hope that it will be useful,
+//    but WITHOUT ANY WARRANTY; without even the implied warranty of
+//    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//    GNU Lesser General Public License for more details.
+//
+//    You should have received a copy of the GNU Lesser General Public License
+//    along with Nfield.SDK.  If not, see <http://www.gnu.org/licenses/>.
+
+using Nfield.Models;
+using Nfield.SDK.Models;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Nfield.SDK.Services
+{
+    /// <summary>
+    /// Service for managing manual test surveys
+    /// </summary>
+    public interface INfieldSurveyManualTestsService
+    {
+        /// <summary>
+        /// Get manual tests of a specific survey (normally just one)
+        /// </summary>
+        Task<IQueryable<SurveyManualTest>> GetSurveyManualTestsAsync(string surveyId);
+
+        /// <summary>
+        /// Get manual test of a specific survey
+        /// </summary>
+        Task<SurveyManualTest> GetSurveyManualTestAsync(string surveyId, string manualTestId);
```

Comment:
I don't understand these endpoints. What are they returning? I would expect that we have 1 endpoint without any parameters to return all the manual test surveys in the domain, and another endpoint that for a given survey id, you return the manual test survey

Reply:
The endpoint to return all manual test surveys in the domain is the GetAsync method in NfieldManualTestsService class. This class is for the one with the given surveyId => GetSurveyManualTestsAsync(string surveyId); It returns an array with normally just one test survey. Then we have another endpoint that returns just the one test survey => GetSurveyManualTestAsync(string surveyId, string manualTestId);    You could say that the last one is not needed. But Team Yellow preferred it that way. It is conform the REST standards.

Reply:
But why we need an endpoint for this? If they want to get the manual test survey, they can use the regular GetSurvey endpoint using the manualTestId as surveyId

Reply:
The GetSurvey endpoint does not return the OriginalSurveyId though. But on the other hand, you would have to know the OriginalSurvey to use the new (maybe not needed) endpoint.